### PR TITLE
Feat migrate on memfs

### PR DESCRIPTION
**Pull request by evilebottnawi**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Migrate on `memfs` + rewrite test from `output: { path: '/' }` on `output: { path: __dirname }` because using `/` with `memfs` is invalid

### Breaking Changes

Yes

### Additional Info

No

### CODE REVIEW

File: src/utils/setupOutputFileSystem.js [hunk 1]
```diff
@@ -15,38 +15,31 @@ export default function setupOutputFileSystem(compiler, context) {
...
+    // Todo remove when we drop webpack@4 support
     if (typeof fs.join !== 'function') {
       // very shallow check
       throw new Error('Invalid options: options.fs.join() method is expected');
     }
 
-    // eslint-disable-next-line no-param-reassign
-    compiler.outputFileSystem = fs;
-    fileSystem = fs;
-  } else if (isMemoryFs) {
```
Comment (evilebottnawi) [inline src/utils/setupOutputFileSystem.js]: I don't know why it is here :confused:

Comment (evilebottnawi) [reply src/utils/setupOutputFileSystem.js]: Also it doesn't not work when you use multi compiler mode

**Comment by evilebottnawi** [comment]
What is going on with azure last days :confused:

**Comment by evilebottnawi** [comment]
Regression on webpack side:
- before `{ output: { path: '/' } }` doesn't throw error and you can emit files in existing directory
- now `{ output: { path: '/' } }` throw error what directory already exists

**Comment by hiroppy** [comment]
Sorry but I rebased from the latest next branch for audit so please resolve the conflict 🙏

**Comment by evilebottnawi** [comment]
Done in master

**Pull request closed without merging**